consensus/dummy/dynamic_fee_window.go

```diff
@@ -0,0 +1,92 @@
+// (c) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package dummy
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/coreth/params"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+var ErrInsufficientDynamicFeeWindowLength = errors.New("insufficient length for dynamic fee window")
```

Comment:
nit starting the err name with `ErrDynamicFeeWindow` would be better when searching for errors related to the dynamic fee window, so you could rename this to `ErrDynamicFeeWindowInsufficientLength`